### PR TITLE
Migrate to latest ODB changes: workflows and offsets in every sequence step

### PR DESCRIPTION
**Pull request by rpiaggio**

### CODE REVIEW

File: modules/server_new/src/clue/scala/observe/common/ObsQueriesGQL.scala [hunk 1]
```diff
@@ -17,9 +17,12 @@ object ObsQueriesGQL:
 
   @GraphQL
   trait ActiveObservationIdsQuery extends GraphQLOperation[ObservationDB]:
+    // TODO The ODB API doesn't provide a way to filter ready observations,
+    // so we filter by accepted proposals for now.
+    // Revise this when the API supports it OR we start getting obersvations from the scheduler.
```
Comment (cquiroz) [inline modules/server_new/src/clue/scala/observe/common/ObsQueriesGQL.scala:22]: Wouldnt the obs workflow have a state saying we are ready?

File: modules/server_new/src/main/scala/observe/server/tcs/TcsSouth.scala [hunk 2]
```diff
@@ -223,48 +224,30 @@ object TcsSouth {
...
-    )
-    val gwod3  = none.flatMap(_ =>
-      StepConfig.science.andThen(StepConfig.Science.guiding).getOption(stepConfig)
-    )
-    val gwod4  = none.flatMap(_ =>
-      StepConfig.science.andThen(StepConfig.Science.guiding).getOption(stepConfig)
-    )
+    val guiding: StepGuideState = telescopeConfig.guiding
+
+    val gwp1   = none.map(_ => guiding)
```
Comment (cquiroz) [inline modules/server_new/src/main/scala/observe/server/tcs/TcsSouth.scala:236]: Wouldn't the `map` never be called?

**Review by cquiroz — APPROVED** [review]
LGTM

File: modules/server_new/src/clue/scala/observe/common/ObsQueriesGQL.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/server_new/src/clue/scala/observe/common/ObsQueriesGQL.scala:22]: Yes, but:
1) There's no way to set an observation as `Ready` yet.
2) There's no way to filter `Ready` observations.

File: modules/server_new/src/main/scala/observe/server/tcs/TcsSouth.scala (hunk 2 above)
Comment (rpiaggio) [reply modules/server_new/src/main/scala/observe/server/tcs/TcsSouth.scala:236]: Exactly. I don't know why code was this way, I supposed it is a reminder for when we work on the other instruments. I tried to respect that.

Comment (cquiroz) [reply modules/server_new/src/main/scala/observe/server/tcs/TcsSouth.scala:236]: @jluhrs why's this?

**Pull request merged**